.github/workflows/worker-deploy.yml

```diff
@@ -41,9 +49,117 @@ jobs:
           secrets: |
             SUPABASE_URL
             SUPABASE_KEY
+            ${{ secrets.KERNEL_PUBLIC_KEY && secrets.KERNEL_PUBLIC_KEY != '' && 'KERNEL_PUBLIC_KEY' || '' }}
+
         env:
           SUPABASE_URL: ${{ secrets.SUPABASE_URL }}
           SUPABASE_KEY: ${{ secrets.SUPABASE_KEY }}
+          KERNEL_PUBLIC_KEY: ${{ secrets.KERNEL_PUBLIC_KEY }}
+
+      - name: Update manifest.json worker url
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const path = require('path');
+
+            const manifestPath = path.resolve("${{ github.workspace }}", './manifest.json');
+            const manifest = JSON.parse(fs.readFileSync(manifestPath, 'utf8'));
+
+            manifest["homepage_url"] = "${{ steps.wrangler_deploy.outputs.deployment-url }}";
+
+            const updatedManifest = JSON.stringify(manifest, null, 2);
+            fs.writeFileSync(manifestPath, updatedManifest);
+            console.log('Updated manifest:', updatedManifest);
+
+      - name: Get GitHub App token
+        if: env.APP_ID != '' && env.APP_PRIVATE_KEY != ''
+        uses: tibdex/github-app-token@v1.7.0
+        id: get_installation_token
+        with:
+          app_id: ${{ env.APP_ID }}
+          private_key: ${{ env.APP_PRIVATE_KEY }}
+
+      - name: Format manifest using Prettier
+        shell: bash
+        run: |
+          bun add -DE prettier
+          bun prettier --write .
+
+      - name: Update manifest.json
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ steps.get_installation_token.outputs.token || github.token }}
+        run: |
+          git add "manifest.json"
+          echo "Changed files:"
+          echo "$(git diff-index --cached --name-only HEAD)"
+          if [ -n "$(git diff-index --cached --name-only HEAD)" ]; then
+            node -e "
+              const fs = require('fs');
+              const path = require('path');
+              const github = require('@actions/github');
+              const manifestPath = path.relative('.', 'manifest.json');
+
+              async function pushChanges() {
+                const octokit = github.getOctokit(process.env.GITHUB_TOKEN);
+
+                const context = github.context;
+                const owner = context.repo.owner;
+                const repo = context.repo.repo;
+                const ref = 'heads/${{ github.ref_name }}';
```

Comment:
Never tried with the `sha` directly, I'll test that.

I wish we could do that with `git` CLI directly but it doesn't allow to commit as `ubiquity-os bot` being the author because the token for a bot is not usable through the CLI as far as I know. If we would commit as `github-actions` using the `GITHUB_TOKEN` yes all of this would take two command lines. Do we still want the bot to auth the commits @0x4007 ?

Reply:
I think it's possible to make it appear as the bot is committing but you still use `GITHUB_TOKEN`: https://github.com/actions/create-github-app-token?tab=readme-ov-file#configure-git-cli-for-an-apps-bot-user

Reply:
Nice, didn't find that before, it seems quite recent, will definitely give this a try.

Reply:
@whilefoo So the problem with just setting the email and username in the Git environment is that the commits are not having a `Verified` status, which is why I was using `Octokit` to make sure they were in the first place. What I did is use the package mentioned in what you linked to get the token (because the package I was using is very old and is using a deprecated version of node) but instead of manually committing the manifest I found a package that handles the Verified badge (which most likely does what I was doing under the hood but at least now we don't have to take care of it ourselves). C.f. screenshot
<img width="1289" alt="image" src="https://github.com/user-attachments/assets/f4f4951c-56c5-4f5e-a41e-539ae485889c" />